streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientHandler.scala

```diff
@@ -36,6 +36,19 @@ object FlinkClientHandler {
     }
   }
 
+  def triggerSavepoint(savepointRequest: TriggerSavepointRequest): SavepointResponse = {
+    savepointRequest.executionMode match {
+      case ExecutionMode.LOCAL => LocalSubmit.triggerSavepoint(savepointRequest)
+      case ExecutionMode.REMOTE => RemoteSubmit.triggerSavepoint(savepointRequest)
+      case ExecutionMode.YARN_APPLICATION => YarnApplicationSubmit.triggerSavepoint(savepointRequest)
+      case ExecutionMode.YARN_SESSION => YarnSessionSubmit.triggerSavepoint(savepointRequest)
+      case ExecutionMode.YARN_PER_JOB | ExecutionMode.YARN_SESSION => YarnPerJobSubmit.triggerSavepoint(savepointRequest)
+      case ExecutionMode.KUBERNETES_NATIVE_SESSION => KubernetesNativeSessionSubmit.triggerSavepoint(savepointRequest)
+      case ExecutionMode.KUBERNETES_NATIVE_APPLICATION => KubernetesNativeApplicationSubmit.triggerSavepoint(savepointRequest)
+      case _ => throw new UnsupportedOperationException(s"Unsupported ${savepointRequest.executionMode} Submit ")
+    }
+  }
+
```

Comment:
Hi @RocMarshal , `FlinkClientHandler` is a bad case for code design, it has too many `match  ... case `.

It can be refactored after this PR, would you like to work on it? 

cc @wolfboys

Reply:
Indeed, as @1996fanrui you said. The current design logic makes the code inevitably bloated.
We can start with a new Issue to sort out the desired effect of the refactoring, the problems at this stage, and the best way to refactor.
CC @wolfboys

Reply:
> Hi @RocMarshal , `FlinkClientHandler` is a bad case for code design, it has too many `match ... case `.
> 
> It can be refactored after this PR, would you like to work on it?
> 
> cc @wolfboys

Indeed, this part of the code needs to be refactored, which can be done in the next PR

Reply:
> Indeed, as @1996fanrui you said. The current design logic makes the code inevitably bloated. We can start with a new Issue to sort out the desired effect of the refactoring, the problems at this stage, and the best way to refactor. CC @wolfboys

yes. we can refactor it in next PR